Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx

```diff
@@ -1139,185 +1136,216 @@ void qSlicerMarkupsModuleWidget::updateMaximumScaleFromVolumes()
     {
-    selectedItem->setCheckState(Qt::Unchecked);
+    item->setCheckState(selectedState);
     }
-  // disable editing so that a double click won't bring up an entry box
-  selectedItem->setFlags(selectedItem->flags() & ~Qt::ItemIsEditable);
-  int selectedIndex = d->columnIndex("Selected");
-  if (d->activeMarkupTableWidget->item(m,selectedIndex) == nullptr ||
-      (d->activeMarkupTableWidget->item(m,selectedIndex)->checkState() != selectedItem->checkState()))
+  if (isNewItem)
     {
-    d->activeMarkupTableWidget->setItem(m,selectedIndex,selectedItem);
+    d->activeMarkupTableWidget->setItem(controlPointIndex, column, item);
     }
 
   // locked
-  QTableWidgetItem* lockedItem = new QTableWidgetItem();
-  // disable checkable
-  lockedItem->setData(Qt::CheckStateRole, QVariant());
-  lockedItem->setFlags(lockedItem->flags() & ~Qt::ItemIsUserCheckable);
-  // disable editing so that a double click won't bring up an entry box
-  lockedItem->setFlags(lockedItem->flags() & ~Qt::ItemIsEditable);
-  if (markupsNode->GetNthControlPointLocked(m))
+  column = d->columnIndex("Locked");
+  item = d->activeMarkupTableWidget->item(controlPointIndex, column);
+  isNewItem = false;
+  if (!item)
     {
-    lockedItem->setData(Qt::UserRole, QVariant(true));
-    lockedItem->setData(Qt::DecorationRole, d->SlicerLockIcon);
+    item = new QTableWidgetItem();
+    // disable checkable
+    item->setData(Qt::CheckStateRole, QVariant());
+    item->setFlags(item->flags() & ~Qt::ItemIsUserCheckable);
+    // disable editing so that a double click won't bring up an entry box
+    item->setFlags(item->flags() & ~Qt::ItemIsEditable);
+    isNewItem = true;
     }
-  else
+  bool locked = markupsNode->GetNthControlPointLocked(controlPointIndex);
+  if (isNewItem || item->data(Qt::UserRole).toBool() != locked)
     {
-    lockedItem->setData(Qt::UserRole, QVariant(false));
-    lockedItem->setData(Qt::DecorationRole, d->SlicerUnlockIcon);
+    item->setData(Qt::UserRole, QVariant(locked));
+    item->setData(Qt::DecorationRole, locked ? d->SlicerLockIcon : d->SlicerUnlockIcon);
     }
-  int lockedIndex = d->columnIndex("Locked");
-  if (d->activeMarkupTableWidget->item(m,lockedIndex) == nullptr ||
-      d->activeMarkupTableWidget->item(m,lockedIndex)->data(Qt::UserRole) != lockedItem->data(Qt::UserRole))
+  if (isNewItem)
     {
-    d->activeMarkupTableWidget->setItem(m,lockedIndex,lockedItem);
+    d->activeMarkupTableWidget->setItem(controlPointIndex, column, item);
     }
 
   // visible
-  QTableWidgetItem* visibleItem = new QTableWidgetItem();
-  // disable checkable
-  visibleItem->setData(Qt::CheckStateRole, QVariant());
-  visibleItem->setFlags(visibleItem->flags() & ~Qt::ItemIsUserCheckable);
-  // disable editing so that a double click won't bring up an entry box
-  visibleItem->setFlags(visibleItem->flags() & ~Qt::ItemIsEditable);
-  if (markupsNode->GetNthControlPointVisibility(m))
+  column = d->columnIndex("Visible");
```

Comment:
Anticipating such string may be translated, I suggest to introduce index like what is done here https://github.com/Slicer/Slicer/blob/f6eb38811b642f629a34b2ded5e393f6c0d417d1/Base/QTCore/qSlicerExtensionsManagerModel.cxx#L105-L150

Reply:
OK